pgjdbc/src/main/java/org/postgresql/PGProperty.java

```diff
@@ -855,10 +857,20 @@ public String getDescription() {
    * @param properties properties to take actual value from
    * @return evaluated value for this connection parameter
    */
-  public @Nullable String get(Properties properties) {
+  public @Nullable String getOrDefault(Properties properties) {
     return properties.getProperty(name, defaultValue);
   }
 
+  /**
+   * Returns the value of the connection parameters according to the given {@code Properties}
+   *
+   * @param properties properties to take actual value from
+   * @return evaluated value for this connection parameter or null
+   */
+  public @Nullable String get(Properties properties) {
+    return properties.getProperty(name);
```

Comment:
This looks like a severe breakage of the public API.
Previously `.get` did take the default value into the consideration, while now `get` ignores the default value.

I believe code like `PGProperty.PG_HOST.get(properties)` is used in the wild, and it looks like it would change the behavior after this change.

Reply:
I will revert it. Thanks!

Reply:
Well, I should have figured it out myself that get() could be used by external libraries.

Can I make rollback commit to the same PR here and it can be merged again? (or new PR is required)

What could be the name of another method:
- getWithoutDefault()
- getNoDefault()
- ?

Reply:
if the only issue with this PR is that get should be reverted we can just fix that as per PR # 2644
After contemplating this a bit, I'm not sure of the value of using getOrDefault ?
@vlsi do you have any other issues with the PR ?

Reply:
Well, we can probably keep `getOrDefault` so it explicitly conveys the behavior.

We can keep `get()` which would delegate to `getOrDefault`, and mark it as `@Deprecated` (without intention to remove the method), so the clients who use the methods could either migrate to the "new" `getOrDefault` or `getWithoutDefault`, etc.

I think `getOrNull(...)` would be better alternative than `getWithoutDefault` and `getIgnoringDefault`.
AFAIK, `Properties` (and `Hashtable`) does not support `null` values, and `orNull` in the method name gives an explicit warning that the method can return `null`.

I don't like `getNoDefault`.
Yet another alternative could be `getIgnoringDefault`, however, I think `getOrNull` is better for "get value or return null if it is missing in the properties".

Reply:
@davecramer 

> if the only issue with this PR is that get should be reverted we can just fix that as per PR # 2644

yes, that's good

@vlsi 

>  I think `getOrNull `is better ...

I like it.